producer.go

```diff
@@ -11,6 +11,41 @@ import (
 // ProduceFunc is a callback which is run by the producer on every iteration.
 type ProduceFunc func(*Writer) error
 
+// LastModFunc is a function to return local data last modification time.
+type LastModFunc func(context.Context) (time.Time, error)
+
+// ProducerState holds current state of producer.
+type ProducerState struct {
```

Comment:
K, I checked something and it works. There is no need to make `ProducerState` public.

1. rename to `producerState`
2. and embed as value not as pointer

```go
type Producer struct {
  producerState
  // ...
}
```

Reply:
here's what I have done to confirm:

```go
// main.go
package main

import "x/internal"

func main() {
	x := new(internal.X)
	x.Inc()
	x.Inc()
	println(x.N()) // => 2
}
```

```go
// internal/types.go

package internal

type X struct {
	y
}

type y struct {
	n int
}

func (y *y) Inc() {
	y.n++
}

func (y *y) N() int {
	return y.n
}
```

Reply:
hmmm, interesting.